libtransmission/announcer.c

```diff
@@ -1434,6 +1508,29 @@ static void on_scrape_done(tr_scrape_response const* response, void* vsession)
         }
     }
 
+    /* Maybe reduce the number of torrents in a multiscrape req */
+    if (multiscrape_too_big(response->errmsg))
+    {
+        char const* url = response->url;
+        int* multiscrape_max = &tr_announcerGetScrapeInfo(announcer, url)->multiscrape_max;
+
+        if (*multiscrape_max == response->row_count)
+        {
+            const int n = MAX(TR_MULTISCRAPE_MIN, *multiscrape_max - TR_MULTISCRAPE_STEP);
+            if (*multiscrape_max != n)
+            {
+                char* host = NULL;
+                if (tr_urlParse(url, strlen(url), NULL, &host, NULL, NULL))
+                {
+                    tr_logAddNamedInfo(host, "Reducing multiscrape max to %d", n);
```

Comment:
The message could be misleading here if the same host is used with different scheme and/or port and/or path. Not sure how frequent that is... Why don't we just use `url` (or at least more than just host) for name?

Reply:
That's a good point. I hesitated to do that because these URLs can have personal announce IDs, which should probably be kept out of logs.

Reply:
In a55d303997c84f993861cdb7152de833056a92cf I've added the scheme and port back in, but stopped short of including the path for the reason above. Thoughts?